lib/ansible/modules/cloud/smartos/nictagadm.py

```diff
@@ -0,0 +1,238 @@
+    name:
+        description:
+            - NIC tag name.
+        required: true
+    mac:
+        description:
+            - MAC Address to attach the NIC tag to when not using an etherstub. mac and etherstub are mutually exclusive.
+        required: false
+        default: None
+    etherstub:
+        description:
+            - Specifies that the NIC tag will be attached to a created etherstub. etherstub is mutually exclusive with both mtu, and mac.
+        required: false
+        default: false
+    mtu:
+        description:
+            - MTU size of the NIC tag. mtu and etherstub are mutually exclusive.
+        required: false
+        default: None
+    force:
+        description:
+            - When State.Absent is set this switch will use the -f parameter and delete the NIC tag regardless of existing VMs
+        default: false
+    state:
+        description:
+            - Create or delete a SmartOS NIC tag.
+        required: false
+        default: "present"
+        choices: [ "present", "absent" ]
+'''
+
+EXAMPLES = '''
+- name: Create 'strage0' on '00:1b:21:a3:f5:4d'
+  nictagadm: name=storage0 mac=00:1b:21:a3:f5:4d mtu=9000 state=present
+
+- name: Remove 'storage0' NIC tag
+  nictagadm: name=storage0 state=absent force=false
+'''
+
+RETURN = '''
+name:
+    description: NIC tag name
+    returned: always
+    type: string
+    sample: storage0
+mac:
+    description: MAC Address that the nic tag was attached to.
+    returned: always
+    type: string
+    sample: 00:1b:21:a3:f5:4d
+etherstub:
+    description: specifies if the NIC tag will create and attach to an etherstub.
+    returned: always
+    type: boolean
+    sample: False
+mtu:
+    description: specifies which MTU size was passed during the nictagadm add command. mtu and etherstub are mutually exclusive.
+    returned: always
+    type: int
+    sample: 1500
+force:
+    description: Shows if -f was used during the deletion of a NIC tag
+    returned: always
+    type: boolean
+    sample: False
+state:
+    description: state of the target
+    returned: always
+    type: string
+    sample: present
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+import re
+
+
+class NICTAG(object):
+
+    def __init__(self, module):
+        self.module = module
+
+        self.name = module.params['name']
+        self.mac = module.params['mac']
+        self.etherstub = module.params['etherstub']
+        self.mtu = module.params['mtu']
+        self.force = module.params['force']
+        self.state = module.params['state']
+
+    def is_valid_mac(self):
+        if re.match("[0-9a-f]{2}([:])[0-9a-f]{2}(\\1[0-9a-f]{2}){4}$", self.mac.lower()):
+            return True
+        else:
+            return False
+
+    def nictag_exists(self):
+        cmd = [self.module.get_bin_path('nictagadm', True)]
+
+        cmd.append('exists')
+        cmd.append(self.name)
+
+        (rc, dummy, dummy) = self.module.run_command(cmd)
+
+        if rc == 0:
```

Comment:
Could be shortened to `return rc == 0` if you're not doing any further processing other than returning.